### PR TITLE
4.x : Workspace Refactoring : Feature to add a datasource definition

### PR DESCRIPTION
**Pull request by deepakrkris**
Create a datasource node in the workspace graph

- [X] - define a DataSource Class
- [X] - add a DataSourceDefinition loopback model
- [X] - add workspace task to add a Datasource node to a workspace graph
- [X] - extend connector to create a datasource
- [X] - add gherkin cucumber test case to create a datasource
- [X] - use Dirty Chai
- [X] - move graph modules in datamodel to sub directory 'graph'

### CODE REVIEW

File: common/models/data-source-definition.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+'use strict';
+
+module.exports = function(DataSourceDefinition) {
+  /**
+   * Creates a DataSource definition.
+   *
+   * @class DataSourceDefinition
+   */
+  DataSourceDefinition.on('dataSourceAttached', function(eventData) {
+    var connector = DataSourceDefinition.getConnector();
```
Comment (superkhau) [inline common/models/data-source-definition.js]: move this down to line 21 -- where it's used

File: component/tasks.js [hunk 2]
```diff
@@ -13,6 +14,11 @@ class Tasks {
     var model = new Model(workspace, modelId, modelDef);
     cb(null, modelDef);
   }
+  addDataSource(id, datasource, cb) {
+    var workspace = this;
+    var ds = new DataSource(workspace, id, datasource);
+    cb(null, datasource);
```
Comment (superkhau) [inline component/tasks.js]: we don't want to pass back ds?

File: test/acceptance-tests/workspace-devtime/step-definitions/create-datasource.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      };
+      DataSourceDefinition.create(datasource, {}, function(err, data) {
+        if (err) return next(err);
+        testsuite.expectedDs = datasource;
+        next();
+      });
+    });
+
+  this.Then(/^the datasource definition is created$/, function(next) {
+    var expect = chai.expect;
```
Comment (superkhau) [inline test/acceptance-tests/workspace-devtime/step-definitions/create-datasource.js]: Move out to shared helper -- see https://github.com/strongloop/loopback-connector-kv-redis/blob/master/test/helpers/expect.js

**Comment by superkhau** [comment]
In addition to all the above? Are you intending to use ES6? ie. const + let instead of var?

File: common/models/data-source-definition.js (hunk 1 above)
Comment (deepakrkris) [inline common/models/data-source-definition.js]: comment

File: component/tasks.js (hunk 2 above)
Comment (deepakrkris) [inline component/tasks.js]: we cannot instantiate without a variable

File: common/models/data-source-definition.js (hunk 1 above)
Comment (deepakrkris) [inline common/models/data-source-definition.js]: test

**Pull request merged**